storey/redis_driver.py

```diff
@@ -202,10 +202,13 @@ def convert_redis_value_to_python_obj(cls, value):
             ret = False
         else:
             # if value is a number, convert type to int / float
-            no_sign_value = value[1:] if value[0] in ("-", "+") else value
-            if no_sign_value.replace(".", "", 1).isdigit():
-                ret = int(value) if int(float(value)) == float(value) else float(value)
-            else:
+            try:
+                ret = float(value)
+                try:
+                    ret = int(value)
+                except ValueError:
```

Comment:
I wonder, what is a float, not nan/inf, and cannot be converted to int?

Reply:
>>> int('1.2')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '1.2'

Reply:
But it's already converted to float first...

Reply:
Oh ok, `int()` takes the original value, not the result of `float()`.

Reply:
But then, doesn't it make more sense to call `int()` first, and only call `float()` if the former fails?

Reply:
both options are correct but i think float-then-int is slightly better from perf perspective if we count the number of exceptions for each data-type:

float-then-int:
type      exception-count
float        0
int           1
string     1

int-then-float:
type      exception-count
float        1
int           0
string     2

Reply:
The count is a bit off but good enough for me.